### PR TITLE
Improve XML validation

### PR DESCRIPTION
**Pull request by ziadoz**
- Added checks to see if XML/XSD files are readable (useful to know incase permissions are wrong).
- Fix bug where DOMDocument truncates invalid data by moving the saving of formatted/prettified XML after validation has been successful (this has occurred in the live environment numerous times).
- Added more unit tests.

### CODE REVIEW

File: src/Xml.php [hunk 1]
```diff
@@ -35,12 +36,8 @@ public static function validate($filePath)
     {
         if (! file_exists($filePath)) {
             throw new XmlException('XML file missing');
-        }
-
-        // domdocument dies silently when given a big (1.7GB) file, though known to cope with 892Mb
-        // @todo: look at using xmlreader instead @see: https://gist.github.com/tentacode/5934634 for some examples
-        if (filesize($filePath) > 1000000000) { // 1Gb
```
Comment (gabriel403) [inline src/Xml.php]: why have you removed this?

Comment (ziadoz) [inline src/Xml.php]: Ah, I didn't mean to. I'll bring it back.

Comment (ziadoz) [inline src/Xml.php]: @gabriel403 I've fixed that.

**Comment by gabriel403** [comment]
:+1:

**Pull request merged**